contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -135,4 +187,27 @@ abstract contract TeleporterUpgradeable is ITeleporterReceiver {
      * This function should be overridden by contracts that inherit from this contract.
      */
     function _checkTeleporterUpgradeAccess() internal virtual;
+
+    /**
+     * @dev Returns the Teleporter messenger used to send Teleporter messages,
+     * and checks that the Teleporter messenger is not paused.
+     *
+     * By default returns the latest Teleporter messenger, but can be overriden to
+     * return a Teleporter messenger of specific version.
+     */
+    function _getTeleporterMessenger()
+        internal
+        view
+        virtual
+        returns (ITeleporterMessenger)
+    {
+        ITeleporterMessenger teleporter = teleporterRegistry
+            .getLatestTeleporter();
+        require(
```

Comment:
Should we allow for sending via non-latest unpaused versions if available?

Reply:
Are you thinking if the `latestVersion` is paused, keep going down the versions to find one that is not paused to send with? That was my initial thought, but I want to stay away from it for the default implementation. Dapps might only be receiving from the latest version, and if they send with a older version they wouldn't be able to receive these messages. Also it becomes less clear which version the dapp sends each Teleporter version with.

Reply:
I think the issue with not going down the list is: If we upgrade from say, v2 to v3, but v3 breaks, we have no way to use v2 again until an out-of-band warp-message is sent to add v2 as v4.

Reply:
Yup a new Teleporter version would needed to registered, but I think that is preferable to default to sending on an older Teleporter version, and leading to messages that can't be received.

Reply:
I don't think it's too much to ask to have dapp owners be aware of what versions are and are not available on other chains. In fact, I think it's a requirement if you're running a cross-chain app, because new teleporter versions can arrive on different chains at different times. If we wanted to be explicit about it, we could have a separate function call for `enableFallback` or something, which would have to be called while the latest version is paused.

I'm very wary against locking people out of using a teleporter version they've said their contract can use. They can then only use their contract again once the validators submit a new version with an out-of-band warp message, which they are not guaranteed to do, let alone on any particular timeline.

Reply:
In this case, the contract owners should be able to unpause the version though right? As long as that's the case, I'm fine with pausing the latest version having the effect of being unable to send messages. The origin transaction would revert, so it wouldn't take any effect until it's unpaused.

Reply:
Discussed offline that anything extra here is probably over-engineering.